src/main/java/seedu/duke/ui/TextUi.java

```diff
@@ -6,13 +6,89 @@
+    public static final String MODULE_ALREADY_EXISTS = "A module with that name already exists...";
+    public static final String ESTIMATED_WORKING_TIME = "Estimated Working Time: ";
+
+    /**.
+     * For deleteCommand
+     */
+    public static final String DELETE_MESSAGE = " has been deleted.";
+
+    /**.
+     * For exitCommand
+     */
+    public static final String READY_EXIT = "I am ready to exit *_*";
+
+    /**.
+     * For listCommand
+     */
+    public static final String LIST_MESSAGE_TOP = "Ok! Here are the task(s) in your list:";
+    public static final String EMPTY_LIST = "(empty)";
+
+    /**.
+     * For markCommand
+     */
+    public static final String MARK_MESSAGE_TOP = "Nice! I have marked this task as completed!";
+    public static final String UNMARK_MESSAGE_TOP = "Ok! I have marked this task for you as uncompleted!";
+    public static final String ICON_UNCOMPLETED = "( )";
+    public static final String ICON_COMPLETED = "(X)";
+
+    /**.
+     * For command result
+     */
+    public static final String ARRAYLIST_RESULT = "ArrayList";
+    public static final String STRING_RESULT = "String";
+
+    /**.
+     * For exceptions
+     */
+    public static final String ERROR_NO_SUCH_MODULE = "Sorry, no such module exists ._.";
+    public static final String ERROR_NO_SUCH_TASK = "Sorry, no such task exists ._.";
+    public static final String ERROR_PARSE_FAILED = "This parse failed 0_0";
+    public static final String ERROR_UNKNOWN_COMMAND = "Sorry, I don't understand the following command:";
+    public static final String ERROR_UNSUPPORTED_RESULT_TYPE = "Sorry, I don't understand the result format:";
+
+    /**.
+     * For parsers
+     */
+    public static final String TASK_NAME = "taskName";
+    public static final String TASK_DESCRIPTION = "taskDescription";
+    public static final String TASK_WORKING_TIME = "estimatedWorkingTime";
+    public static final String MODULE_CODE = "moduleCode";
+    public static final String MODULE_DESCRIPTION = "moduleDescription";
+    public static final String TASK_NUMBER = "taskNumber";
+    public static final String FLAG = "flag";
+    public static final String TASK_INDEX = "taskIndex";
+    public static final String COMPLETED_FLAG = "/c";
+    public static final String UNCOMPLETED_FLAG = "/u";
+    public static final String ARGUMENT = "arguments";
+    public static final String COMMAND_WORD = "commandWord";
+    public static final String EXIT_COMMAND_WORD = "exit";
+    public static final String ADD_COMMAND_WORD = "add";
+    public static final String DELETE_COMMAND_WORD = "del";
+    public static final String LIST_COMMAND_WORD = "list";
+    public static final String MARK_COMMAND_WORD = "mark";
+
+    /**.
+     * General Strings
+     */
+    public static final String STRING = "String";
+    public static final String INDENT = "    ";
+    public static final String NULL_STRING = "";
```

Comment:
Maybe we should move all of this stuff to a class like `duke.util.StringConstants`? Putting it in TextUi seems weird because most of these strings have nothing to do with the TextUi functionality

Reply:
What about we put the messages in TextUi and other strings like "String", "mark" go into StringConstants?

Reply:
I support Yikai's point of view. Put the enums to a distinct class can keep the TextUI compact.